docker_images/sklearn/app/pipelines/tabular_classification.py

```diff
@@ -1,18 +1,34 @@
+import json
+import warnings
+from pathlib import Path
 from typing import Dict, List, Union
 
 import joblib
 from app.pipelines import Pipeline
-from huggingface_hub import cached_download, hf_hub_url
+from huggingface_hub import snapshot_download
 
 
 DEFAULT_FILENAME = "sklearn_model.joblib"
 
 
 class TabularDataPipeline(Pipeline):
     def __init__(self, model_id: str):
-        self.model = joblib.load(
-            open(cached_download(hf_hub_url(model_id, DEFAULT_FILENAME)), "rb")
-        )
+        cached_folder = snapshot_download(repo_id=model_id)
+        try:
+            with open(Path(cached_folder) / "config.json") as f:
+                # this is the default path for configuration of a scikit-learn
+                # project. If the project is created using `skops`, it should have
+                # this file.
+                config = json.load(f)
+            model_file = config["sklearn"]["model"]["file"]
+        except Exception as e:
+            print(e)
+            # If for whatever reason we fail to detect requirements of the project,
+            # we install the latest scikit-learn.
+            # TODO: we should log this, or warn or something.
+            model_file = DEFAULT_FILENAME
```

Comment:
For better error handling/messaging I think instead of a `try`/`Exception`, checking for existence of file and respective config keys might be better. Up to you!

Reply:
For the end user it doesn't matter why it fails, we're not showing them any warnings ATM. I'm not sure how to propagate warnings or messages here, that's why I put the TODO above this line. When we do that, then we can catch different exception classes and do the right warning (doesn't have to be an if/else statement for this purpose). Also: https://docs.python.org/3.10/glossary.html#term-eafp